compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -514,6 +514,20 @@ OMR::ARM64::TreeEvaluator::istoreEvaluator(TR::Node *node, TR::CodeGenerator *cg
    return commonStoreEvaluator(node, TR::InstOpCode::strimmw, 4, cg);
    }
 
+// also handles astore, astorei
+TR::Register *
+OMR::ARM64::TreeEvaluator::astoreEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Compilation *comp = cg->comp();
+   bool isCompressedClassPointerOfObjectHeader = TR::Compiler->om.generateCompressedObjectHeaders() &&
+         (node->getSymbol()->isClassObject() ||
+         (node->getSymbolReference() == comp->getSymRefTab()->findVftSymbolRef()));
+   TR::MemoryReference *tempMR = new (cg->trHeapMemory()) TR::MemoryReference(node, isCompressedClassPointerOfObjectHeader ? 4 : 8, cg);
+   TR::InstOpCode::Mnemonic op = isCompressedClassPointerOfObjectHeader ? TR::InstOpCode::strimmw : TR::InstOpCode::strimmx;
+
+   return commonStoreEvaluator(node, TR::InstOpCode::strimmw, 4, cg);
```

Comment:
You need to use `op` here.

Reply:
Oops. Thanks for pointing it out. Fixed the code.